### DISCUSSION TITLE
How do I set a max length of user-visible texts for a WYSIWYG field?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#12308] [Archive]
I saw the following post about setting a max length of a WYSIWYG field, and the answer suggests to use custom API hooks.

https://github.com/directus/directus/discussions/9014

However, this is the length of all chars, including HTML tags added by the WYSIWYG field, like `<p>`, `<strong>` and so on.

These tags are not visible to users of the CMS, therefore should not bother them.

For example, if a WYSIWYG  field has a note "Please enter no more than 5 chars", a user may enter "abcde", which should be valid, but in fact it is not because the wrapping `<p>` and `</p>` take another 7 chars, and 7 + 5 = 12.

Hence if the custom API hook introduces a max length of 5, there will be an exception thrown. 

I am wondering if there is a way to get the user-visible text, which doesn't include HTML tags and is just "abcde" in the above example?

Thanks!

**Comment by rijkvanzanten** [comment]
There's no util available out of the box to strip the HTML tags from a string in a custom hook. That being said, I believe there's a couple npm packages available to do this, or you could even do a "dumb" regex to ignore the html tags completely

**Reply by powerseed** [reply]
Got it. Thanks for the reply!
